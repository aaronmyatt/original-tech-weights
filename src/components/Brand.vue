<template>
    <h3 class="brand">{{ brandName }}</h3>
</template>

<script>
export default {
  name: 'Brand',
  props: ['brandName'],
};
</script>

<style scoped>
</style>
