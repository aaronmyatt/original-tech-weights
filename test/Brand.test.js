import { shallow } from 'vue-test-utils'
import Brand from '../src/components/Brand'

describe('Brand', () => {
  let cmp

  beforeEach(() => {
    cmp = shallow(Brand, {
        propsData: {
            brandName: 'Test',
        }
    })
  })

  it('has the expected html structure', () => {
    expect(cmp.element).toMatchSnapshot()
  })

  describe('Props', () => {

      it('renders the given brand name', () => {
          expect(cmp.find('.brand').text()).toBe(cmp.vm.brandName)
      })

  })
})
