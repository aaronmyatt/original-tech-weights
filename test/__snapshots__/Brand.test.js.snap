// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brand has the expected html structure 1`] = `
<h3
  class="brand"
>
  Test
</h3>
`;
